packages/experimental/src/experimental-list/collapsible-list/index.tsx

```diff
@@ -135,6 +164,12 @@ export const ExperimentalCollapsibleList: React.FC< CollapsibleListProps > = ( {
 		} );
 	};
 
+	// This allows for an extra render cycle that adds the maxHeight back in before the exiting transition.
+	// This way the exiting transition still works correctly.
+	useEffect( () => {
+		setTransitionCollapsed( isCollapsed );
+	}, [ isCollapsed ] );
+
```

Comment:
As mentioned in the comment, this is a slightly hacky way to allow us to remove `maxHeight` when a transition is finished, but add the `maxHeight` back in just before the transition start again.
In order for the CSS transition to work, the maxHeight should already be present when the new `maxHeight` is updated.